src/GovUk.Education.ExploreEducationStatistics.Common/Services/PostgreSqlRepository.cs

```diff
@@ -0,0 +1,99 @@
+#nullable enable
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using GovUk.Education.ExploreEducationStatistics.Common.Model;
+using GovUk.Education.ExploreEducationStatistics.Common.Requests;
+using GovUk.Education.ExploreEducationStatistics.Common.Services.Interfaces;
+using Microsoft.EntityFrameworkCore;
+using Npgsql;
+using NpgsqlTypes;
+using JsonSerializer = System.Text.Json.JsonSerializer;
+
+namespace GovUk.Education.ExploreEducationStatistics.Common.Services;
+
+public class PostgreSqlRepository : IPostgreSqlRepository
+{
+    public async Task<TResponse?> GetJsonbFromPath<TDbContext, TRowId, TResponse>(
```

Comment:
this could potentially be `private` for now. But I can definitely see a case where you would use it. Happy for you to leave it as either. Just wanted to point it out :)

Reply:
Yup going to leave it public for when I implement your suggested validation for checking that the candidate keys are valid candidates!

Reply:
sounds good!